storage/shard.go

```diff
@@ -54,6 +55,23 @@ var (
 	errShardAlreadyClosed = errors.New("shard already closed")
 )
 
+type dbShardMetrics struct {
+	numberOfImmutableSegments tally.Gauge
+	numberOfLoadedSegments    tally.Gauge
+	numberOfUnloadedSegments  tally.Gauge
+	unload                    tally.Counter
```

Comment:
Might want to separate this into two counters: `unloadSuccess` and `unloadErrors`